portfolio/src/main/webapp/paragraph.js

```diff
@@ -0,0 +1,67 @@
+/**
+ * Adds a heading with a paragraph to each picture.
+ */
+function addDescription(directory) {
+    let imagePathDescriptionMap = getImageDescriptionMap(directory);
+
+    generateDiv("content");
+
+    let count = 1;
+    for (const [imagePath, description] of Object.entries(imagePathDescriptionMap)) {
+        generateHTMLLayout(imagePath, description, count, "description");
+        count++;
+    };
+}
+
+/**
+ * Gets the images to be displayed.
+ * @param {string} directory  directory to the images.
+ * @return {array of string array} a map from image path to image description
+ */
+function getImageDescriptionMap(directory) {
+    let mapping = {
+        "images/sibing-1.jpg": "This picture was taken accidentally by my friend two years ago. However it becomes one of my favorite photo since at that time I was still quite slim.",
+        "images/swag.jpg": "Me & Welcome Packge with Google!",
+        "images/welcomePackage.jpg": "Welcome package from Google!",
+        "images/sibing-2.jpg": "My favorite ID photo, since I was truly slim at that time."
+    }
+
+    return mapping;
+}
+
+/**
+ * Creates a "div" that contain images and descriptions.
+ * @param {string} elementId id of the element where the "div" places.
+ */
+function generateDiv(elementId) {
+    let div = document.createElement("div");
+    let id = document.createAttribute("id");
+    id.value = "description";
+    div.setAttributeNode(id);
+
+    let element = document.getElementById(elementId);
+    element.appendChild(div);
+}
+
+/**
+ * Generates the layout in the html.
+ * @param {string} image path to the image.
+ * @param {string} description description to the image.
+ * @param {number} i paragraph id.
```

Comment:
If i is paragraph id, maybe rename it to paragraphId ?

Reply:
Fixed